account_financial_risk/models/account_invoice.py

```diff
@@ -7,32 +7,37 @@
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
+    def risk_exception_msg(self):
+        self.ensure_one()
+        partner = self.partner_id.commercial_partner_id
+        exception_msg = ""
+        if partner.risk_exception:
+            exception_msg = _("Financial risk exceeded.\n")
+        elif partner.risk_invoice_open_limit and (
+                (partner.risk_invoice_open + self.amount_total) >
+                partner.risk_invoice_open_limit):
+            exception_msg = _(
+                "This invoice exceeds the open invoices risk.\n")
+        # If risk_invoice_draft_include this invoice included in risk_total
+        elif not partner.risk_invoice_draft_include and (
+                partner.risk_invoice_open_include and
+                (partner.risk_total + self.amount_total) >
+                partner.credit_limit):
+            exception_msg = _(
+                "This invoice exceeds the financial risk.\n")
+        return exception_msg
+
     @api.multi
     def action_invoice_open(self):
         if self.env.context.get('bypass_risk', False):
             return super(AccountInvoice, self).action_invoice_open()
         for invoice in self.filtered(lambda x: x.type in (
                 'out_invoice', 'out_refund') and x.amount_total_signed > 0.0):
-            partner = invoice.partner_id.commercial_partner_id
-            exception_msg = ""
-            if partner.risk_exception:
-                exception_msg = _("Financial risk exceeded.\n")
-            elif partner.risk_invoice_open_limit and (
-                    (partner.risk_invoice_open + invoice.amount_total) >
-                    partner.risk_invoice_open_limit):
-                exception_msg = _(
-                    "This invoice exceeds the open invoices risk.\n")
-            # If risk_invoice_draft_include this invoice included in risk_total
-            elif not partner.risk_invoice_draft_include and (
-                    partner.risk_invoice_open_include and
-                    (partner.risk_total + invoice.amount_total) >
-                    partner.credit_limit):
-                exception_msg = _(
-                    "This invoice exceeds the financial risk.\n")
+            exception_msg = invoice.risk_exception_msg()
             if exception_msg:
                 return self.env['partner.risk.exceeded.wiz'].create({
```

Comment:
If we do `return`, this will abort massive confirmation batch.

Reply:
This return isn't modify in this PR

Reply:
Yes, but this is not the expected behavior, isn't it? Or batch confirm call other method?

Reply:
This is called one to one in `invoice_confirm()`:
https://github.com/odoo/odoo/blob/9094dd0c8279a8eef582cd5daedefbcfc4af8a21/addons/account/wizard/account_invoice_state.py#L22

Reply:
OK